parso/python/errors.py

```diff
@@ -779,62 +779,68 @@ def message(self):
             return "Generator expression must be parenthesized"
 
     def is_issue(self, node):
-        first_arg = node.children[0]
-        if first_arg.type == 'argument' \
-                and first_arg.children[1].type in _COMP_FOR_TYPES:
-            # e.g. foo(x for x in [], b)
-            return len(node.children) >= 2
-        else:
-            arg_set = set()
-            kw_only = False
-            kw_unpacking_only = False
-            is_old_starred = False
-            # In python 3 this would be a bit easier (stars are part of
-            # argument), but we have to understand both.
-            for argument in node.children:
-                if argument == ',':
-                    continue
+        arg_set = set()
+        kw_only = False
+        kw_unpacking_only = False
+        is_old_starred = False
+        # In python 3 this would be a bit easier (stars are part of
+        # argument), but we have to understand both.
+        for argument in node.children:
+            if argument == ',':
+                continue
 
-                if argument in ('*', '**'):
-                    # Python < 3.5 has the order engraved in the grammar
-                    # file.  No need to do anything here.
-                    is_old_starred = True
-                    continue
-                if is_old_starred:
-                    is_old_starred = False
-                    continue
+            if argument in ('*', '**'):
+                # Python < 3.5 has the order engraved in the grammar
+                # file.  No need to do anything here.
+                is_old_starred = True
+                continue
+            if is_old_starred:
+                is_old_starred = False
+                continue
 
-                if argument.type == 'argument':
-                    first = argument.children[0]
-                    if first in ('*', '**'):
-                        if first == '*':
-                            if kw_unpacking_only:
-                                # foo(**kwargs, *args)
-                                message = "iterable argument unpacking " \
-                                          "follows keyword argument unpacking"
-                                self.add_issue(argument, message=message)
+            if argument.type == 'argument':
+                first = argument.children[0]
+                if (
+                    argument.children[1].type in _COMP_FOR_TYPES
+                    and len(node.children) >= 2
+                ):
+                    if self._normalizer.version >= (3, 7):
+                        return True
+                    elif len(node.children) == 2 and node.children[1] == ",":
+                        # trailing comma allowed until 3.7
+                        pass
```

Comment:
Sorry for the noise, but I feel like this extra case just makes it more complicated without an extra benefit.

Would you mind just ignoring it again? :) Sorry, but I really didn't know that this was only < 3.7. I thought it was a general issue.

Reply:
If you think that this is more correct, I'm fine with that as well, but IMO it really helps nobody to have this distinction for Python 3.5/3.6.

Reply:
I agree, will revert